spec/integration/views_test_spec.rb

```diff
@@ -0,0 +1,151 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :system do
+  before(:each) do
+    @profile1 = User.create(name: 'Nabeel',
+                            photo: 'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg',
+                            bio: 'Teacher from Mexico.',
+                            post_counter: 0)
+    @profile2 = User.create(name: 'Muhammad',
+                            photo: 'https://as2.ftcdn.net/v2/jpg/02/14/74/61/1000_F_214746128_31JkeaP6rU0NzzzdFC4khGkmqc8noe6h.jpg',
+                            bio: 'Teacher from Poland.',
+                            post_counter: 0)
+    @jin = User.create(name: 'Ahmed',
+                       photo: 'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg',
+                       bio: 'Teacher from Mexico.',
+                       post_counter: 0)
+  end
+  before(:each) do
+    @post1 = Post.create(Title: 'title 1',
+                         Text: 'post content 1',
+                         CommentsCounter: 0,
+                         LikesCounter: 0,
+                         author_id: @jin.id)
+    @post2 = Post.create(Title: 'title 2',
+                         Text: 'post content 2',
+                         CommentsCounter: 0,
+                         LikesCounter: 0,
+                         author_id: @jin.id)
+    @post3 = Post.create(Title: 'title 3',
+                         Text: 'post content 3',
+                         CommentsCounter: 0,
+                         LikesCounter: 0,
+                         author_id: @jin.id)
+    @post4 = Post.create(Title: 'title 4',
+                         Text: 'post content 4',
+                         CommentsCounter: 0,
+                         LikesCounter: 0,
+                         author_id: @jin.id)
+    @comment1 = Comment.create(author_id: @profile1.id, post_id: @post1.id, Text: 'Hello')
+    @comment2 = Comment.create(author_id: @profile1.id, post_id: @post1.id, Text: 'What is Up')
+    @like1 = Like.create(author_id: @profile1.id, post_id: @post1.id)
+  end
+  describe 'user index' do
+    before(:each) do
+      visit users_path
+    end
+    it 'Display the user details' do
+      expect(page).to have_content(@profile1.name)
+      expect(page).to have_content(@profile2.name)
+    end
+
+    it 'Display each user profile picture' do
+      expect(page).to have_xpath("//img[contains(@src,'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg')]")
+    end
+
+    it 'Number of posts' do
+      expect(page).to have_content('Number of posts: 0')
+    end
+
+    it 'Redirect to user page' do
+      click_link 'Nabeel'
+      expect(page).to have_current_path("/users/#{@profile1.id}")
+    end
+  end
+
+  describe 'Show user' do
+    before(:each) do
+      visit user_path(@jin.id)
+    end
```

Comment:
- [ ] According to project requirements it should be Tests that assert you can see the **profile picture** of all other users in **user show page**. Please fix to make your test work as expected 😄